app/models/result.rb

```diff
@@ -120,4 +121,16 @@ def check_for_nil_marks
     end
     true
   end
+
+  def create_marks
+    @assignment = self.submission.assignment
+    @marks_map = Hash.new
+    @assignment.get_criteria.each do |criterion|
+      mark = criterion.marks.find_or_create_by(result_id: self.id)
```

Comment:
Just create - there won't be anything to find. Also use `id` instead of `self.id`; self is redundant in this case.